app/uk/gov/hmrc/smassetsfrontend/services/AssetCacheService.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.smassetsfrontend.services
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.FileIO
+import play.api.Logger
+import play.api.cache.AsyncCacheApi
+import play.api.libs.ws.WSClient
+import uk.gov.hmrc.smassetsfrontend.config.AppConfig
+
+import java.io.{File, FilenameFilter}
+import java.util.zip.ZipFile
+import javax.inject.Inject
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class AssetCacheService @Inject()(client: WSClient, config: AppConfig, cache: AsyncCacheApi)(implicit ec: ExecutionContext, mat: Materializer) {
+  private val logger = Logger(this.getClass)
+  private val failedDownloads = mutable.Map[String, String]()
+
+  def getAsset(version:String): Future[Option[ZipFile]] = cache.getOrElseUpdate(version)(downloadIfMissing(version))
+
+
+  def downloadIfMissing( version: String): Future[Option[ZipFile]] = {
+    val maybeFile = config.cacheDir.resolve(s"assets-frontend-$version.zip").toFile
```

Comment:
Just wondering if we need the `cacheDir`, i.e. could we just rely on the in-memory `AsyncCacheApi`? It would mean we'd have to download all the assets again after restarting, but the implementation would be a lot simpler. e.g. here could the file be partially downloaded?